mmv1/products/networkservices/Gateway.yaml

```diff
@@ -0,0 +1,126 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Gateway'
+base_url: 'projects/{{project}}/locations/global/gateways'
+create_url: 'projects/{{project}}/locations/global/gateways?gatewayId={{name}}'
+self_link: 'projects/{{project}}/locations/global/gateways/{{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  Gateway represents the configuration for a proxy, typically a load balancer.
+  It captures the ip:port over which the services are exposed by the proxy,
+  along with any policy configurations. Routes have reference to to Gateways
+  to dictate how requests should be routed by this Gateway.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.gateways'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+import_format: ['projects/{{project}}/locations/global/gateways/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_gateway_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-gateway'
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_gateway_advanced'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-gateway'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    url_param_only: true
+    description: |
+      Short name of the Gateway resource to be created.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'selfLink'
+    description: |-
+      Server-defined URL of this resource.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Time the AccessPolicy was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
+    description: |
+      Time the AccessPolicy was updated in UTC.
+    output: true
+  - !ruby/object:Api::Type::KeyValuePairs
+    name: 'labels'
+    required: false
```

Comment:
It's not necessary to have `required:false` here and in several other places. If you prefer to keep them I'm fine with that too.

Reply:
Done. I removed from all.